snapcraft/plugins/_ros/rospack.py

```diff
@@ -0,0 +1,146 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import logging
+import subprocess
+import tempfile
+
+from snapcraft.internal import common, repo
+from snapcraft import formatting_utils
+
+logger = logging.getLogger(__name__)
+
+
+class Rospack:
+    def __init__(
+        self,
+        *,
+        ros_distro,
```

Comment:
can you add type annotations for this and maybe docstrings too?

Reply:
Of course, done.